### DISCUSSION TITLE
Background Notifications

### DISCUSSION THREAD

**Discussion by bergmorten** [tauri-apps/tauri#8423] [Q&A]
Hi, I've not yet tried Tauri, but looking for an alternative to Electron since background notifications is not supported. Background notification are notification received by the OS will the Tauri app is closed. Does Tauri support background notifications, e.g. Firebase Messaging with a service-worker?

Electron does not have support for the required PushManger since the embedded Chrome does not have this capability,

**Comment by ranjit41771** [comment]
even I am looking for background notification solution.

**Comment by gordongrech** [comment]
Hi All. I have managed to achieve this using plugins for both Android and iOS, I have not tried (and don't need this) on desktop. I approached this as follows: 

- Create a plugin
- Write native notification services in both Kotlin and Swift. 
- Push messages back to Rust. 

Last part was the most difficult for me. There may be a better way however I could not find any. Documentation stipulates that events can be triggered by native code and listened for in Javascript but I needed these in Rust. In the end, I created an interop layer in C using a static C library for iOS and Uniffi bindings with a dynamic library for Kotlin. 

It's not easy but doable. Hope this helps.

**Reply by osdiab** [reply]
How did you end up building this? Would be helpful to see an example!

**Comment by osdiab** [comment]
Someone seems to be working on contributing this, not sure the current status though.

https://github.com/tauri-apps/tauri/issues/11651
